docs/developers/optimistic-oracle-integration.md

```diff
@@ -0,0 +1,104 @@
+
+The requestor can offer a reward for the proposer who responds to a price request, but the reward value is set to `0` in this example.
+
+The price proposer posts a bond along with their price, which will be refunded if the price is not disputed, or if a dispute is resolved in the proposer's favor. Otherwise, this bond is used to pay the final fee to the DVM and pay a reward to a successful disputer.
+
+## Running the Demo
+
+1. Ensure that you have followed all the prerequisite setup steps [here](developers/setup.md).
+2. Run a local Ganache instance (i.e. not Kovan/Ropsten/Rinkeby/Mainnet) with `yarn ganache-cli --port 9545`
+3. In another window, migrate the contracts by running the following command:
+
+```bash
+yarn truffle migrate --reset --network test
+```
+
+1. To deploy the `OptimisticDepositBox` [contract](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/financial-templates/demo/OptimisticDepositBox.sol) and go through a simple user flow, run the following demo script from the root of the repo:
+
+```bash
+yarn truffle exec ./packages/core/scripts/demo/OptimisticDepositBox.js --network test
+```
+
+You should see the following output:
+
+```
+1. Deploying new OptimisticDepositBox
+  - Using wETH as collateral token
+  - Pricefeed identifier for ETH/USD is whitelisted
+  - Collateral address for wETH is whitelisted
+  - Deployed an OptimisticOracle
+  - Deployed a new OptimisticDepositBox
+
+
+2. Minting ERC20 to user and giving OptimisticDepositBox allowance to transfer collateral
+  - Converted 10 ETH into wETH
+  - User's wETH balance: 10
+  - Increased OptimisticDepositBox allowance to spend wETH
+  - Contract's wETH allowance: 10
+
+
+3. Depositing ERC20 into the OptimisticDepositBox
+  - Deposited 10 wETH into the OptimisticDepositBox
+  - User's deposit balance: 10
+  - Total deposit balance: 10
+  - User's wETH balance: 0
+
+
+4. Withdrawing ERC20 from OptimisticDepositBox
+  - Submitted a withdrawal request for 10000 USD of wETH
+  - Proposed a price of 2000000000000000000000 ETH/USD
+  - Fast-forwarded the Optimistic Oracle and Optimistic Deposit Box to after the liveness window so we can settle.
+  - New OO time is [fast-forwarded timestamp]
+  - New ODB time is [fast-forwarded timestamp]
+  - Executed withdrawal. This also settles and gets the resolved price within the withdrawal function.
+  - User's deposit balance: 5
+  - Total deposit balance: 5
+  - User's wETH balance: 5
+```
+
+## Explaining the Contract Functions
+
+It is worthwhile to examine the `OptimisticDepositBox` [contract code](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/financial-templates/demo/OptimisticDepositBox.sol) to see how it interacts with the UMA protocol.
```

Comment:
Can you mention somewhere that this is a mock OptimisticOracle contract and perhaps point users to the mainnet/kovan ones in case they want to extend the example to integrate with production addresses?

Reply:
We actually use the real `OptimisticOracle` contract here, not a mock, though it is deployed on a local testnet. I can add some links to the production addresses, though.

Reply:
Oh gotcha, that makes sense.